### ISSUE TITLE
Bug: In-line disabling of the strict rule requires the comment to be the first line of the file (regression in ESLint v10)

### ISSUE THREAD

**Issue by Standard8** [eslint/eslint#20451]
### Environment

Node version: v25.3.0
npm version: 11.7.0
Local ESLint version: 10.0.0-rc.0
Global ESLint version: N/A
Operating System: Mac 26.2

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```js
export default [
  {
    files: ["*.js"],
    languageOptions: {
      sourceType: "script",
    },
    rules: {
      strict: ["error", "global"],
    }
  }
]
```
</details>

```js
/* Copyright message, or comment explaning why strict is disabled. */
/* eslint-disable strict */

function foo() {
  console.log(foo);
}
```

### What did you expect to happen?

No errors

### What actually happened?

```
  1:1  error    Use the global form of 'use strict'                                        strict
  2:1  warning  Unused eslint-disable directive (no problems were reported from 'strict')
```

### Link to Minimal Reproducible Example

https://github.com/Standard8/eslint-strict-mode-comment-location

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

If the `/* eslint-disable strict */` line is on the very first line of the file, then it works. However, if it is not on the very first line then it fails, even if it is before any code.

This prevents situations like shown in the example, where we want to have a copyright message at the start of the file, or even a comment as to why strict mode is being disabled in the file.

This is a regression in the Eslint 10.0.0 series (I checked the alpha & beta and they failed also). Eslint 9.39.2 works fine.

**Comment by mdjermanovic** [comment]
Thanks for the issue! I can reproduce this with the latest prerelease version 10.0.0-rc.1 as well.

This happens because the `strict` rule in this case reports the `Program` node, which used to start from the first statement, but as of ESLint v10 the `Program` node starts from the first character in the source code (relevant change: https://github.com/eslint/js/pull/677), so an `eslint-disable` directive placed anywhere else doesn't apply to it.

Since in this use case the `strict` rule is effectively disabled in the entire file, you could replace `/* eslint-disable strict */`  with `/* eslint strict: off */`. Would that work for you?

**Comment by Standard8** [comment]
The problem with `/* eslint strict: off */` is that developers might not realise that is needed. The default disable capabilities for editors appear to be to use the `eslint-disable` options, though they do add it on the first line. At a minimum I think this would need documenting in various places, but I do feel it is a bit non-obvious and contrary to how the rest of the disable system works.

**Comment by mdjermanovic** [comment]
Then maybe we could update the `strict` rule to report the same range as in v9. @eslint/eslint-tsc thoughts?

**Comment by fasttime** [comment]
> Then maybe we could update the `strict` rule to report the same range as in v9. [@eslint/eslint-tsc](https://github.com/orgs/eslint/teams/eslint-tsc) thoughts?

I think that's a good idea 👍

**Comment by nzakas** [comment]
Agree, I think this is a bug as a result of the `Program` range change and we should restore the original behavior.

**Comment by ntnyq** [comment]
I'd like to work on this.

**Comment by mdjermanovic** [comment]
@ntnyq thanks! Would you be able to submit a PR today as we'd like to release another RC with this fix?

**Comment by ntnyq** [comment]
> [@ntnyq](https://github.com/ntnyq) thanks! Would you be able to submit a PR today as we'd like to release another RC with this fix?

Yes, the PR has been created.

**Comment by mdjermanovic** [comment]
The fix for this issue is included in ESLint v10.0.0-rc.2.

https://eslint.org/blog/2026/01/eslint-v10.0.0-rc.2-released/
